resourcestore/fileresourcestore.go

```diff
@@ -0,0 +1,466 @@
+package resourcestore
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/cluster"
+	"github.com/weaveworks/flux/image"
+	"github.com/weaveworks/flux/policy"
+	"github.com/weaveworks/flux/resource"
+)
+
+type resourceWithOrigin struct {
+	resource   resource.Resource
+	configFile *ConfigFile // only set if the resource came from a configuration file
+}
+
+type fileResourceStore struct {
+	ctx             context.Context
```

Comment:
Didn't notice this before -- contexts should not be stored in a struct, they need to be passed for each operation that may be subject to a context. This means you'll have to add them to the ResourceStore methods, I suspect.

Reply:
That's fine, will do